client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/generic.tsx

```diff
@@ -0,0 +1,105 @@
+import { isDomainProduct, isPlan, isTitanMail } from '@automattic/calypso-products';
+import { Button } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import ThankYouV2 from 'calypso/components/thank-you-v2';
+import ThankYouProduct from 'calypso/components/thank-you-v2/product';
+import { useSelector } from 'calypso/state';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getSelectedSiteSlug, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getDomainPurchaseTypeAndPredicate } from '../../utils';
+import ThankYouDomainProduct from '../products/domain-product';
+import ThankYouPlanProduct from '../products/plan-product';
+import { ThankYouTitanProduct } from '../products/titan-product';
+import getDefaultFooterDetails from './content/get-default-footer-details';
+import getDomainFooterDetails from './content/get-domain-footer-details';
+import getTitanFooterDetails from './content/get-titan-footer-details';
+
+interface GenericThankYouProps {
+	purchases: ReceiptPurchase[];
+	emailAddress?: string;
+}
+
+export default function GenericThankYou( { purchases, emailAddress }: GenericThankYouProps ) {
+	const siteSlug = useSelector( getSelectedSiteSlug );
+	const siteId = useSelector( getSelectedSiteId );
+	const currentRoute = useSelector( getCurrentRoute );
+	const [ , predicate ] = getDomainPurchaseTypeAndPredicate( purchases );
+	const filteredPurchases = purchases.filter( ( purchase ) => {
+		return ! isDomainProduct( purchase ) || predicate( purchase );
+	} );
```

Comment:
When users purchase a domain _registration_, we'll have two separate purchase items in the list, one for domain registration and one for domain mapping. These lines of code ensure we filter out the redundant item that should not be listed in the frontend.